### PR TITLE
Inlining cost model: don't count :invokes that `throw`

### PR DESCRIPTION
**Pull request by timholy**
The throw is not typically part of the standard runtime cost of the function. This should fix the performance concern with #20005, which turns out to be an inlining difference.

### CODE REVIEW

File: base/inference.jl [hunk 1]
```diff
@@ -4649,7 +4649,12 @@ function statement_cost(ex::Expr, src::CodeInfo, mod::Module, params::InferenceP
         end
         return plus_saturate(argcost, params.inline_nonleaf_penalty)
     elseif head == :foreigncall || ex.head == :invoke
```
Comment (yuyichao) [inline base/inference.jl]: Unrelated but `head == :invoke` ? (Or `===`)

**Comment by timholy** [comment]
@nanosoldier `runbenchmarks(ALL, vs = ":master")`

**Comment by nanosoldier** [comment]
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/22732#issuecomment-314055916):
```
NanosoldierError: failed to run benchmarks against comparison commit: failed process: Process(`sudo cset shield -e su nanosoldier -- -c ./benchscript.sh`, ProcessExited(1)) [1]
```
Logs and partial data can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/a0ae254bb9e657c180c06997167fa89709cde129/5f9b2bd_vs_1e95e1d/report.md)
cc @jrevels

**Comment by timholy** [comment]
Something seems wrong with nanosoldier, possibly https://github.com/JuliaCI/BaseBenchmarks.jl/issues/83. I've checked locally; there are 5 reproducible regressions, the worst of which is 20%. There are also 5 reproducible improvements, of which some are nearly 10x. Since 10x beats 1.2x, I'm merging.

**Pull request merged**